src/boot/efi/util.c

```diff
@@ -91,10 +91,36 @@ EFI_STATUS efivar_set(const EFI_GUID *vendor, const CHAR16 *name, const CHAR16 *
 }
 
 EFI_STATUS efivar_set_uint(const EFI_GUID *vendor, CHAR16 *name, UINTN i, BOOLEAN persistent) {
-        CHAR16 str[32];
+        assert_cc(sizeof(UINTN) == sizeof(UINT64) || sizeof(UINTN) == sizeof(UINT32));
+
+        return sizeof(UINTN) == sizeof(UINT32) ? efivar_set_uint32(vendor, name, i, persistent) :
+                                                 efivar_set_uint64(vendor, name, i, persistent);
+}
+
+EFI_STATUS efivar_set_uint32(const EFI_GUID *vendor, CHAR16 *name, UINT32 i, BOOLEAN persistent) {
+        UINT8 buf[4];
+
+        buf[0] = (UINT8)(i >> 0U);
```

Comment:
i'd add & 0xFF here, since C doesn't actually say how precisely to cast down iirc

Reply:
I think it does for unsigned but better be safe.